src/OSPSuite.Core/Domain/Services/SensitivityAnalyses/SensitivityAnalysisRunner.cs

```diff
@@ -50,14 +54,18 @@ public async Task Run(SensitivityAnalysis sensitivityAnalysis, SensitivityAnalys
 
          try
          {
+            var cts = new CancellationTokenSource();
             if (IsRunning)
                throw new OSPSuiteException(Error.CannotStartTwoConcurrentSensitivityAnalyses);
 
-            var options = runOptions ?? new SensitivityAnalysisRunOptions { NumberOfCoresToUse = _coreUserSettings.MaximumNumberOfCoresToUse};
+            if (!_cancellationTokenSources.TryAdd(sensitivityAnalysis, cts)) //this will prevent from running one that is already running
+               throw new OSPSuiteException(Error.CannotStartTwoConcurrentSensitivityAnalyses);
```

Comment:
I don't understand this. We can start two concurrent SA. Not for the same SA object. 
The message should be "Sensitivity Analysis is already running for SAxxx" or sthg like this no?

Reply:
No we can't start two SA, never could.